search/src/Components/upload/react-images-upload/index.js

```diff
@@ -129,11 +130,25 @@ class ReactImageUploadComponent extends React.Component {
                 dataURLs.push(newFileData.dataURL);
                 files.push(newFileData.file);
                 this.setState({pictures: dataURLs, files: files});
+                const updatedFilesName = files.map(file => file.name);
+                this.syncImageErrorMessage(updatedFilesName);
             }
         })
     }
   }
 
+  syncImageErrorMessage(updatedFilesName) {
+    const missingImagesAmount = this.props.images.length - updatedFilesName.length;
+    const missingImages = [...this.props.images].filter(image => !updatedFilesName.includes(image));
+    if (missingImagesAmount == 0 && new Set(missingImages).size === 0) {
+        this.props.handleImageReady(true);
+        this.props.handleErrorMessage("");
+    } else {
+        this.props.handleImageReady(false);
+        this.props.handleErrorMessage(`${missingImagesAmount} ${missingImagesAmount > 1 ? "images" : "image"} missing`, {error_type: "image", missingImages: missingImages});
```

Comment:
I'm not entirely with the separation of concerns here... It feels like you're pushing too much weed-ai-specific specific work onto the upload component. That is, I think it's good if `ReactImageUploadComponent` calls a handler, e.g. `handleUploaded(updatedFiles)` or `handleUploaded(updatedFileNames)`, but the translation of that to an error message should be weed-ai logic, not in react-images-upload code.

What other changes had you made to react-images-upload?

Reply:
You mean move syncImageErrorMessage outside and rename it to handleUploaded?

Reply:
Effectively, yes, but the goal is that ReactImageUploadComponent doesn't deal with things that are not about the image uploading, i.e. it shouldn't care about which files are expected/unexpected.